### DISCUSSION TITLE
Support for Mouse Button (4 and 5) Hotkeys

### DISCUSSION THREAD

**Discussion by jangoergens** [sumatrapdfreader/sumatrapdf#4218] [Ideas]
Allow remapping of Shortcuts to mouse buttons (Left Click, Right Click, Middle Click, Side Button).

Use case:

Command Back: Mouse Button 4 instead of Alt + Left Arrow
Command Forward: Mouse Button 5 instead of Alt + Right Arrow

**Comment by kjk** [comment]
Sadly, not easy to do.

**Reply by jangoergens** [reply]
Unfortunate :( 

Just out of curiosity, what is the reason for the difficulty?

**Comment by GitHubRulesOK** [comment]
@jangoergens 
Mouse[s!] with programmable 3-16 buttons usually come with their own key allocations 
If you did not download/install such drivers then [X-Mouse (MBC)](https://www.highrez.co.uk/downloads/XMouseButtonControl.htm) is a very good well respected alternative.

>up to 10 'layers' of different button configurations which you can switch between using hot-keys or mouse buttons.
 XMBC only supports up to 5 mouse buttons because it relies on, and intercepts, the **Windows mouse message queue which also only supports 5 buttons.**

**Reply by jangoergens** [reply]
Thanks for the suggestion! I will use that as an alternative until (or if) it gets implemented natively.
